### PR TITLE
suite: add support for building ffmpeg with libjxl de/encoding support

### PR DESCRIPTION
**Pull request by woot000**
kinda hacky but it works (both static and shared ffmpeg) for me

### CODE REVIEW

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1093,19 +1093,28 @@ if [[ $libavif = y ]] && {
...
     log -q "git.submodule" git submodule update --init --recursive
-    do_cmake global -D{BUILD_TESTING,JPEGXL_ENABLE_{BENCHMARK,MANPAGES,OPENEXR,SKCMS,EXAMPLES}}=OFF \
-        -DJPEGXL_{BUNDLE_GFLAGS,FORCE_SYSTEM_BROTLI,STATIC}=ON \
-        -DJPEGXL_FORCE_SYSTEM_HWY=OFF
-    do_ninja
-    do_install tools/{c,d}jxl.exe bin-global/
+    [[ $jpegxl = y ]] || extracommands=("-DJPEGXL_ENABLE_TOOLS=OFF")
+    do_cmakeinstall global -D{BUILD_TESTING,JPEGXL_ENABLE_{BENCHMARK,MANPAGES,OPENEXR,SKCMS,EXAMPLES}}=OFF \
+        -DJPEGXL_{BUNDLE_GFLAGS,FORCE_SYSTEM_BROTLI,STATIC}=ON -DJPEGXL_FORCE_SYSTEM_HWY=OFF "${extracommands[@]}"
+    [[ $jpegxl = y ]] && do_install tools/{{c,d}jxl{,_ng},cjpeg_hdr,jxlinfo}.exe bin-global/
```
[the remark this answers is not in the document]
Comment (woot000) [reply build/media-suite_compile.sh:1110]: can't see a reason not to, the individual parts of ```do_cmakeinstall``` are rarely used in the compile script and this does what I expect it to do

File: build/media-suite_compile.sh [hunk 2]
```diff
@@ -1093,19 +1093,28 @@ if [[ $libavif = y ]] && {
...
+    mv -f "$LOCALDESTDIR/lib/libjxl-static.a" "$LOCALDESTDIR/lib/libjxl.a"
+    mv -f "$LOCALDESTDIR/lib/libjxl_dec-static.a" "$LOCALDESTDIR/lib/libjxl_dec.a"
+    [[ $ffmpeg != no ]] && enabled libjxl && \
+        mv -f "$LOCALDESTDIR/lib/libjxl_threads-static.a" "$LOCALDESTDIR/lib/libjxl_threads.a"
+    sed -i 's|__declspec(dllimport)||g' "$LOCALDESTDIR/include/jxl/jxl_export.h"
```
[the remark this answers is not in the document]
Comment (woot000) [reply build/media-suite_compile.sh:1115]: this appears to work for me, adding it

File: build/media-suite_compile.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (1480c1) [reply build/media-suite_compile.sh:1110]: the main reason I asked was that a good amount of the .exes are installed on the `ninja install` phase, but you are manually installing some of them
```bash
-- Installing: D:/mabs/local64/bin-global/cjxl.exe
-- Installing: D:/mabs/local64/bin-global/cjxl_ng.exe
-- Installing: D:/mabs/local64/bin-global/djxl.exe
-- Installing: D:/mabs/local64/bin-global/djxl_ng.exe
-- Installing: D:/mabs/local64/bin-global/cjpeg_hdr.exe
-- Installing: D:/mabs/local64/bin-global/jxlinfo.exe
```

File: build/media-suite_compile.sh [hunk 3]
```diff
@@ -1093,19 +1093,28 @@ if [[ $libavif = y ]] && {
...
-    do_ninja
-    do_install tools/{c,d}jxl.exe bin-global/
+    [[ $jpegxl = y ]] || extracommands=("-DJPEGXL_ENABLE_TOOLS=OFF")
+    do_cmakeinstall global -D{BUILD_TESTING,JPEGXL_ENABLE_{BENCHMARK,MANPAGES,OPENEXR,SKCMS,EXAMPLES}}=OFF \
+        -DJPEGXL_{BUNDLE_GFLAGS,FORCE_SYSTEM_BROTLI,STATIC}=ON -DJPEGXL_FORCE_SYSTEM_HWY=OFF "${extracommands[@]}"
+    [[ $jpegxl = y ]] && do_install tools/{{c,d}jxl{,_ng},cjpeg_hdr,jxlinfo}.exe bin-global/
+    mv -f "$LOCALDESTDIR/lib/libjxl-static.a" "$LOCALDESTDIR/lib/libjxl.a"
+    mv -f "$LOCALDESTDIR/lib/libjxl_dec-static.a" "$LOCALDESTDIR/lib/libjxl_dec.a"
+    [[ $ffmpeg != no ]] && enabled libjxl && \
+        mv -f "$LOCALDESTDIR/lib/libjxl_threads-static.a" "$LOCALDESTDIR/lib/libjxl_threads.a"
```
Comment (1480c1) [inline build/media-suite_compile.sh:1114]: I think this is fine if you move it unconditionally in the case that something else other than ffmpeg uses `libjxl_threads.pc`

File: build/media-suite_compile.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (woot000) [reply build/media-suite_compile.sh:1110]: fair point, using ```do_cmakeinstall``` also automatically installed a couple of unnecessary libraries (libhwy_test.a and libhwy_contrib.a don't seem to be used during the build process of ffmpeg)
i'll change it

**Comment by 1480c1** [comment]
seems good to me, do you want me to rebase and merge or squash?

**Comment by woot000** [comment]
you can squash it to two commits

**Comment by 1480c1** [comment]
merged as c4da880d4a1196e07bb30c4b97e0443d5958f4bb and b984784516ea25b925cbd38757681d659be80681

Thanks

**Pull request closed without merging**